src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,147 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Destination;
+import com.google.sps.data.LatLng;
+import com.google.sps.data.Riddle;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Returns a destination created from user submitted information */
+@WebServlet("/destination-data")
+public class DestinationDataServlet extends HttpServlet {
+
+  // Temporarily stores the destination created by the user
+  public List<Destination> destinations = new ArrayList<>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
+
+    LatLng location =
+        new LatLng.Builder()
+            .withLat(Double.parseDouble(request.getParameter("latitude")))
+            .withLng(Double.parseDouble(request.getParameter("longitude")))
+            .build();
+
+    String city = request.getParameter("city");
+    String description = request.getParameter("description");
+
+    Riddle riddle =
+        new Riddle.Builder()
+            .withPuzzle(request.getParameter("riddle"))
+            .withHint(request.getParameter("hint1"))
+            .withHint(request.getParameter("hint2"))
+            .withHint(request.getParameter("hint3"))
+            .build();
+
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    List<String> levels =
+        Arrays.stream(request.getParameterValues("obscurity"))
+            .filter(level -> level != null)
+            .collect(Collectors.toList());
+    Destination.Obscurity level = convertLevelToEnum(levels);
+
+    /* Retrieves the tags selected by the user as a List of Strings and converts them into a Set of Enum Tags */
+    List<String> tags =
+        Arrays.stream(request.getParameterValues("tag"))
+            .filter(tag -> tag != null)
+            .collect(Collectors.toList());
+    Set<Destination.Tag> checkedTags = convertTagsToEnum(tags);
+
+    Destination d1 =
+        new Destination.Builder()
+            .withName(name)
+            .withLocation(location)
+            .withCity(city)
+            .withDescription(description)
+            .withRiddle(riddle)
+            .withTags(checkedTags)
+            .withObscurity(level)
+            .build();
+
+    destinations.add(d1);
```

Comment:
The idea is that eventually this would write to the datastore, right?  If so, please leave a TODO comment so that it's clear what the ulitmate goal of this code is.

Reply:
yes that is the plan